daemons/fenced/fenced_commands.c

```diff
@@ -916,6 +939,10 @@ build_device_from_xml(xmlNode * msg)
         read_action_metadata(device);
         stonith__device_parameter_flags(&(device->flags), device->id,
                                         device->agent_metadata);
+    } else {
+        device->timer = mainloop_timer_add("get_agent_metadata", 30 * 1000,
+                                           TRUE, get_agent_metadata_cb, device);
+        mainloop_timer_start(device->timer);
```

Comment:
A device may have duplicate registrations, so it may be possible that device->timer already exists at this point. We can guard the add with timer!=NULL and the start with !mainloop_timer_running() (to avoid restarting the timer in that case).